src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -1310,11 +1304,24 @@ void install_signal_handlers() {
   PPC64_ONLY(set_signal_handler(SIGTRAP);)
   set_signal_handler(SIGXFSZ);
   if (!ReduceSignalUsage) {
-    // This is just for early initialization phase. Intercepting the signal here reduces the risk
-    // that an attach client accidentally forces HotSpot to quit prematurely. We skip the periodic
-    // check because late initialization will overwrite it to UserHandler.
-    set_signal_handler(BREAK_SIGNAL, false);
+    // Install BREAK_SIGNAL's handler in early initialization phase, in
+    // order to reduce the risk that an attach client accidentally forces
+    // HotSpot to quit prematurely.
+    // The actual work for handling BREAK_SIGNAL is performed by the Signal
+    // Dispatcher thread, which is created and started at a much later point,
+    // see os::initialize_jdk_signal_support(). Any BREAK_SIGNAL received
+    // before the Signal Dispatcher thread is started is queued up via the
+    // pending_signals[BREAK_SIGNAL] counter, and will be processed by the
+    // Signal Dispatcher thread in a delayed fashion.
+    //
+    // Also note that HotSpot does NOT support signal chaining for BREAK_SIGNAL.
+    // Applications that require a custom BREAK_SIGNAL handler should run with
+    // -XX:+ReduceSignalUsage. Otherwise if libjsig is used together with
```

Comment:
The SIGBREAK handler does not seem to honor `-XX:+AllowUserSignalHandlers`. It is still installed even with `-XX:+AllowUserSignalHandlers`.

Perhaps it can be addressed separately later? As commented in https://bugs.openjdk.org/browse/JDK-8293466, I haven't seen a real use case for +AllowUserSignalHandlers, so I'd vote for deprecating `AllowUserSignalHandlers` for removal in the future.

Reply:
Okay!

Reply:
Thanks. Filed https://bugs.openjdk.org/browse/JDK-8293580.